### ISSUE TITLE
Mean multivariate normal distribution

### ISSUE THREAD

**Issue by espenmei** [kaskr/adcomp#194]
Hi,

I recently came across TMB, thanks a lot for a great program! 

I've been trying to figure out how TMB works, it is really helpful with the example scripts. Now I’ve been trying to figure how to specify means for multivariate normal distributions. Would be great if anyone had some tips for how to specify this!

Sorry if this isn't very clear; my statistics skills are not the best.

For example, if the model has two random coefficients that has a multivariate normal distribution. The random variables are in `PARAMETER_ARRAY(eta)`, their standard deviations in `PARAMETER_VECTOR(sd_eta)` and their correlation in `PARAMETER_VECTOR(rho_eta).` Then I believe I could specify this as:

```
UNSTRUCTURED_CORR_t<Type> nll_eta(rho_eta);
for(int j=0; j<J; j++) {
    nll += VECSCALE(nll_eta,sd_eta)(eta.transpose().col(j));
}
```

This seems to be working. I then tried also to estimate the means of the random variables. The means are in `PARAMETER_VECTOR(gamma)`.  After some trial and error I found this to be working:
```
for(int j=0; j<J; j++) {
    nll += VECSCALE(nll_eta,sd_eta)(eta.transpose().col(j));
    eta(j,0) += gamma(0);
    eta(j,1) += gamma(1);
}
```

I do get the right estimates. But is this a correct way to do this? If I change the order of the arguments it does not work anymore.

```
for(int j=0; j<J; j++) {
    eta(j,0) += gamma(0);
    eta(j,1) += gamma(1);
    nll += VECSCALE(nll_eta,sd_eta)(eta.transpose().col(j));
}
```

Would be very grateful if someone would offer some guidance on this!

Thanks!
Espen

**Comment by skaug** [comment]
Regarding your question, it is custom to define latent random variables to have mean 0, and then use them later as "eta+mean". Btw, you would never assign to the parameter eta. 
I recommend reading the docs http://kaskr.github.io/adcomp/ .

**Comment by espenmei** [comment]
Dear Skaug,

Thank you for the reply. I've tried looking through the docs but could not figure this one out. The reason I wondered is because i wanted to regress "eta" on other fixed and latent variables. Then it seemed convenient to specify this directly in the means of "eta".

But could I do something like this:
```
array<Type> mean_eta(J,2);
for(int j=0; j<J; j++) {
    nll += VECSCALE(nll_eta, sd_eta)(eta.transpose().col(j));
    mean_eta(j,0) = gamma(0);
    mean_eta(j,1) = gamma(1);
}
```

And then use "(eta + mean_eta)" further down in the program? Is that something you would rather recommend?

Thank you very much for the help! If I figure this one out I think I have figured out most of the building blocks I need for the models that I use (thanks to the documentation).

Espen
